### PR TITLE
autoschema rework

### PR DESCRIPTION
**Pull request by yaoshengzhe**
Two commits:

1. return error when schema change fails
2. add schema diffs in schemamanager
  1). Make DiffSchema compare table views.
  2). Add schema diffs in schemamanager. Each schema changes have to change
       table structure and schemanager rejects a sql that does not change
       any table definition.

### CODE REVIEW

File: go/vt/mysqlctl/proto/schema.go [hunk 1]
```diff
@@ -189,6 +189,12 @@ func (sd *SchemaDefinition) ToSQLStrings() []string {
 // generates a report on what's different between two SchemaDefinition
 // for now, we skip the VIEW entirely.
```
Comment (alainjobart) [inline go/vt/mysqlctl/proto/schema.go]: comment needs updating

File: go/vt/mysqlctl/proto/schema.go [hunk 2]
```diff
@@ -189,6 +189,12 @@ func (sd *SchemaDefinition) ToSQLStrings() []string {
...
 func DiffSchema(leftName string, left *SchemaDefinition, rightName string, right *SchemaDefinition, er concurrency.ErrorRecorder) {
```
Comment (alainjobart) [inline go/vt/mysqlctl/proto/schema.go]: probably need a unit test case or two for this? I think we already have some in there somewhere.

File: go/vt/mysqlctl/proto/schema.go (hunk 1 above)
Comment (yaoshengzhe) [inline go/vt/mysqlctl/proto/schema.go]: Done.

File: go/vt/mysqlctl/proto/schema.go (hunk 2 above)
Comment (yaoshengzhe) [inline go/vt/mysqlctl/proto/schema.go]: Done.

**Comment by yaoshengzhe** [comment]
Add more test cases for go/vt/mysqlctl/proto/schema.go and also fix schema.go coding styles based on golint's suggestions.

File: go/vt/mysqlctl/proto/schema.go [hunk 3]
```diff
@@ -67,10 +63,6 @@ func (sd *SchemaDefinition) String() string {
 	return jscfg.ToJSON(sd)
 }
 
-func (sd *SchemaDefinition) SortByReverseDataLength() {
```
Comment (alainjobart) [inline go/vt/mysqlctl/proto/schema.go]: This was supposed to be used by data copiers, so they start with the biggest tables and minimize their run time. But you're right that it's now unused. May need to revive it at some point. :)

File: go/vt/mysqlctl/proto/schema.go [hunk 4]
```diff
@@ -17,10 +16,13 @@ import (
 )
 
 const (
+	// TABLE_BASE_TABLE indicates the table type is a base table.
 	TABLE_BASE_TABLE = "BASE TABLE"
```
Comment (alainjobart) [inline go/vt/mysqlctl/proto/schema.go]: Will have to convert to Camel Case at some point... :(

**Comment by alainjobart** [comment]
LGTM

File: go/vt/mysqlctl/proto/schema.go (hunk 4 above)
Comment (yaoshengzhe) [inline go/vt/mysqlctl/proto/schema.go]: Will do another PR for this since TABLE_BASE_TABLE is referenced by many other places (including internal one)
